Multisig/UI/Settings/OwnerKeyManagement/AddDelegateKeyController.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  AddDelegateKeyController.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 29.11.21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+
+class AddDelegateKeyController {
+
+    weak var presenter: UIViewController?
+    var clientGatewayService = App.shared.clientGatewayService
+
+    private let ownerAddress: Address
+    private let completionHandler: () -> Void
+
+    init(ownerAddress: Address, completion: @escaping () -> Void) {
+        self.ownerAddress = ownerAddress
+        self.completionHandler = completion
+    }
+
+    func start() {
+        // 1. generate a delegate key
+        // 16 bit = 12 words
+        let delegateSeed = Data.randomBytes(length: 16)!
+        let delegateMnemonic = BIP39.generateMnemonicsFromEntropy(entropy: delegateSeed)!
+        let delegatePrivateKey = try! PrivateKey(mnemonic: delegateMnemonic, pathIndex: 0)
+
+        // 2. create 'create delegate' message
+            // keccak(address + str(int(current_epoch // 3600)))
+        let time = String(describing: Int(Date().timeIntervalSince1970) / 3600)
```

Comment:
It seems like not passing time and relying on hours might have some rare edge cases when it might fail. Why did we decide not to send to backend a timestamp to eliminate this risk @DmitryBespalov?